benchmark/hdf5/gen_fbin_file.cpp

```diff
@@ -44,20 +44,30 @@ class Create_FBIN : public Benchmark_hdf5, public ::testing::Test {
     TearDown() override {
     }
 
+    template <bool is_binary>
```

Comment:
This overcomplicates the code.
Please put the new functionality into new functions, such as `fbin_write_binary()` or `fbin_read_binary()`.

Reply:
updated